src/main/java/jenkins/bouncycastle/api/PEMManager.java

```diff
@@ -0,0 +1,400 @@
+        } else if (object instanceof KeyPair) {
+            return ((KeyPair) object).getPrivate();
+        }
+        return null;
+    }
+
+    /**
+     * Obtains raw JCA or BouncyCastle {@link Object} from the read PEM. Depending on the PEM nature or the object
+     * passed to the {@link #PEMManager(Object pemObject)}, the returned object can be one of the following (not
+     * exhaustive list) and any classes that inherit from them:
+     * <ul>
+     * <li><strong>JCA</strong>
+     * <ul>
+     * <li>{@link Certificate}
+     * <li>{@link java.security.cert.CRL}
+     * <li>{@link KeyPair}
+     * <li>{@link PublicKey}
+     * <li>{@link PrivateKey}
+     * </ul>
+     * </ul>
+     * <ul>
+     * <li><strong>Bouncy Castle</strong>
+     * <ul>
+     * <li>{@link org.bouncycastle.asn1.cms.ContentInfo}
+     * <li>{@link org.bouncycastle.jce.spec.ECNamedCurveParameterSpec}
+     * <li>{@link org.bouncycastle.pkcs.PKCS10CertificationRequest}
+     * <li>{@link org.bouncycastle.jce.provider.X509CertificateObject}
+     * <li>{@link org.bouncycastle.x509.X509V2AttributeCertificate}
+     * </ul>
+     * </ul>
+     * 
+     * @return {@link Object} read from the PEM
+     */
+    @CheckForNull
+    public Object getRawObject() {
+        return object;
+    }
+
+    /**
+     * Obtains the fingerprint of the private key in the "ab:cd:ef:...:12" format, which basically is an SHA1 digest
+     * from the key, encoded in hex format.
+     * 
+     * @return private key fingerprint in hex format "ab:cd:ef:...:12", null if the private key could not be obtained
+     * from the current PEM data.
+     * @throws IOException thrown if a problem exists creating the fingerprint
+     */
+    @CheckForNull
+    public String getPrivateKeyFingerprint() throws IOException {
+        PrivateKey key = toPrivateKey();
+        if (key == null) {
+            return null;
+        }
+        return hexEncode(getKeyDigestSHA1(key));
+    }
+
+    /**
+     * Obtains the fingerprint of the public key in the "ab:cd:ef:...:12" format, which basically is an MD5 digest from
+     * the key, encoded in hex format.
+     * 
+     * @return public key fingerprint in hex format "ab:cd:ef:...:12", null if the public key could not be obtained from
+     * the current PEM data.
+     * @throws IOException if a problem exists creating the fingerprint
+     */
+    @CheckForNull
+    public String getPublicKeyFingerprint() throws IOException {
+        PublicKey key = toPublicKey();
+        if (key == null) {
+            return null;
+        }
+        return hexEncode(getKeyDigestMD5(key));
+    }
+
+    /**
+     * Generates an SHA1 digest from a Key object
+     * 
+     * @param k the key to generate the digest from
+     * @return the generated digest
+     * @throws IOException if a problem exists creating the digest
+     * @see <a href="https://docs.oracle.com/javase/7/docs/api/java/security/MessageDigest.html"> JDK Digest
```

Comment:
no. the exception thrown should have the link in the message. No caller of this should ever have to worry *because* the exception should never be thrown according to the Java Language Specification